base/float.jl

```diff
@@ -501,41 +501,21 @@ for Ti in (Int64,UInt64,Int128,UInt128)
         end
     end
 end
-==(x::Float16, y::Union{Int128,UInt128}) = Float64(x)==Float64(y)
-==(x::Union{Int128,UInt128}, y::Float16) = Float64(x)==Float64(y)
-
-<(x::Float16, y::Union{Int128,UInt128}) = Float64(x)<Float64(y)
-<(x::Union{Int128,UInt128}, y::Float16) = Float64(x)<Float64(y)
-
-<=(x::Float16, y::Union{Int128,UInt128}) = Float64(x)<=Float64(y)
-<=(x::Union{Int128,UInt128}, y::Float16) = Float64(x)<=Float64(y)
-
-==(x::Float16, y::Union{Int64,UInt64}) = Float64(x)==Float64(y)
-==(x::Union{Int64,UInt64}, y::Float16) = Float64(x)==Float64(y)
-
-<(x::Float16, y::Union{Int64,UInt64}) = Float64(x)<Float64(y)
-<(x::Union{Int64,UInt64}, y::Float16) = Float64(x)<Float64(y)
-
-<=(x::Float16, y::Union{Int64,UInt64}) = Float64(x)<=Float64(y)
-<=(x::Union{Int64,UInt64}, y::Float16) = Float64(x)<=Float64(y)
-
-==(x::Float16, y::Union{Int16,UInt16}) = Float32(x)==Float32(y)
-==(x::Union{Int16,UInt16}, y::Float16) = Float32(x)==Float32(y)
-
-<(x::Float16, y::Union{Int16,UInt16}) = Float32(x)<Float32(y)
-<(x::Union{Int16,UInt16}, y::Float16) = Float32(x)<Float32(y)
-
-<=(x::Float16, y::Union{Int16,UInt16}) = Float32(x)<=Float32(y)
-<=(x::Union{Int16,UInt16}, y::Float16) = Float32(x)<=Float32(y)
+for op in (:(==), :<, :<=)
+    @eval begin
+        ($op)(x::Float16, y::Union{Int128,UInt128}) = ($op)(Float64(x), Float64(y))
+        ($op)(x::Union{Int128,UInt128}, y::Float16) = ($op)(Float64(x), Float64(y))
 
-==(x::Union{Float16,Float32}, y::Union{Int32,UInt32}) = Float64(x)==Float64(y)
-==(x::Union{Int32,UInt32}, y::Union{Float16,Float32}) = Float64(x)==Float64(y)
+        ($op)(x::Float16, y::Union{Int64,UInt64}) = ($op)(Float64(x), Float64(y))
+        ($op)(x::Union{Int64,UInt64}, y::Float16) = ($op)(Float64(x), Float64(y))
```

Comment:
Can’t you merge this into the Int128 union above?

Reply:
I tried it out now (i.e, method for just `Int128` ),  it says `(Inf16 == typemax(Int64)) = true`. This is because there is typecasting involved. Hence, we need different methods for all @stevengj and @andyferris

Reply:
The implementation is `Float64(x)==Float64(y)` for both types, so only a single method should be required; probably you had some other bug when you tried it.

Reply:
Oh I get what you are saying. I misunderstood your comment for deleting these lines of code. I made the changes now. Sorry for the trouble.